pkg/apis/core/validation/validation.go

```diff
@@ -4422,6 +4426,22 @@ func validateServiceExternalTrafficFieldsValue(service *core.Service) field.Erro
 	return allErrs
 }
 
+// validateServiceInternalTrafficFieldsValue validates InternalTraffic related
+// spec have legal value.
+func validateServiceInternalTrafficFieldsValue(service *core.Service) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if service.Spec.InternalTrafficPolicy != nil &&
+		*service.Spec.InternalTrafficPolicy != core.ServiceInternalTrafficPolicyAll &&
+		*service.Spec.InternalTrafficPolicy != core.ServiceInternalTrafficPolicyPreferLocal &&
+		*service.Spec.InternalTrafficPolicy != core.ServiceInternalTrafficPolicyLocal {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("internalTrafficPolicy"), service.Spec.InternalTrafficPolicy,
+			fmt.Sprintf("InternalTrafficPolicy must be empty, %v, %v or %v", core.ServiceInternalTrafficPolicyAll, core.ServiceInternalTrafficPolicyPreferLocal, core.ServiceInternalTrafficPolicyLocal)))
```

Comment:
nit: this might be a bit cleaner if we use a string set with all the supported values, see usage of `supportedServiceType` as an example:

https://github.com/kubernetes/kubernetes/blob/c5b07f09671f7a0b4d458eb4da64a0b2195bbecf/pkg/apis/core/validation/validation.go#L4247-L4251

Reply:
https://github.com/kubernetes/kubernetes/blob/c5b07f09671f7a0b4d458eb4da64a0b2195bbecf/pkg/apis/core/validation/validation.go#L4150-L4151